core/collateral/engine.go

```diff
@@ -2046,15 +2052,26 @@ func (e *Engine) getFeeTransferRequest(
 		return getAccountLogError(noMarket, systemOwner, types.AccountTypePendingFeeReferralReward)
 	}
 
-	general, err := e.GetAccountByID(e.accountID(noMarket, t.Owner, assetID, types.AccountTypeGeneral))
-	if err != nil {
-		generalID, err := e.CreatePartyGeneralAccount(ctx, t.Owner, assetID)
+	var (
+		general *types.Account
+		err     error
+	)
+	if t.Owner == types.NetworkParty {
+		general, err = e.GetMarketInsurancePoolAccount(marketID, assetID)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("no insurance pool for the market %w", err)
```

Comment:
in what condition is that possible?

Reply:
It shouldn't be, the error was being returned already, I just made the error message more explicit in case we ever encounter it.